src/api/FileActivities.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @flow
+ * @file Helper for the box File Activity API
+ * @author Box
+ */
+
+import Base from './Base';
+import { PERMISSION_CAN_COMMENT, PERMISSION_CAN_VIEW_ANNOTATIONS, ERROR_CODE_FETCH_ACTIVITY } from '../constants';
+import type { BoxItemPermission } from '../common/types/core';
+import type { ElementsXhrError } from '../common/types/api';
+import type { FileActivity, FileActivityTypes } from '../common/types/feed';
+
+// We only show the latest reply in the UI
+const REPLY_LIMIT = 1;
+
+const getFileActivityQueryParams = (
+    fileID: string,
+    activityTypes?: FileActivityTypes[] = [],
+    enableReplies?: boolean = true,
+) => {
+    const baseEndpoint = `/file_activities?file_id=${fileID}`;
+    const hasActivityTypes = !!activityTypes && !!activityTypes.length;
+    const enabledRepliesQueryParam = enableReplies ? `&enable_replies=true&reply_limit=${REPLY_LIMIT}` : '';
+    const activityTypeQueryParam = hasActivityTypes ? `&activity_types=${activityTypes.join()}` : '';
+
+    return `${baseEndpoint}${activityTypeQueryParam}${enabledRepliesQueryParam}`;
+};
+
+class FileActivities extends Base {
+    /**
+     * API URL for file activities
+     *
+     * @param {string} [id] - a box file id
+     * @return {string} base url for files
+     */
+    getUrl(id?: string): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+
+        return `${this.getBaseApiUrl()}${getFileActivityQueryParams(id, undefined, true)}`;
+    }
+
+    /**
+     * API URL for filtered file activities
+     *
+     * @param {string} [id] - a box file id
+     * @param {Array<FileActivityTypes>} activityTypes - optional. Array of File Activity types to filter by, returns all Activity Types if omitted.
+     * @return {string} base url for files
+     */
+    getFilteredUrl(id?: string, activityTypes?: FileActivityTypes[]): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
```

Comment:
should we make `id` required for this function?

and maybe move the `if (!id)` guard to `getActivities` to prevent calls to `this.checkApiCallValidity`?